PYME/config.py

```diff
@@ -508,3 +508,78 @@ def get_init_filename(filename, legacy_scripts_directory=None):
             return fnp
         
     return None
+
+def update_yaml_keys(fn, d):
+    """ 
+    Update the keys in a YAML file without destroying comments.
+
+    TODO: Plays it fast and lose with regex and won't work for many 
+    situations. Currently only works in the case we have a line
+
+    key : value  # optional comment, optional number of spaces
+                 # around the colon, no option for a space at
+                 # the beginning of the line
+    
+    To make this work well, we'll need a full YAML parser 
+    that handles comments. We could either use ruamel.yaml 
+    (https://pypi.org/project/ruamel.yaml/) or write our own.
+
+    Parameters
+    ----------
+    fn : string
+        Path to a yaml file.
+    d : dict
+        key, value pairs of keys/values to update or append to the
+        end of the file
+    """
+    import re
+
+    # Read the yaml file
+    with open(fn) as f:
+        data = f.read()
+
+    # Update the appropriate keys
+    for k, v in d.items():
+        x = re.search('^{}\s*:\s*([^(#|\n|\s)]*)'.format(k),data,flags=re.MULTILINE)
+        if x is None:
+            data += '\n{}: {}'.format(k, v)
+        else:
+            data = re.sub('^{}\s*:\s*([^(#|\n|\s)]*)'.format(k),
+                          '\n{}: {}'.format(k,v),data,flags=re.MULTILINE)
```

Comment:
using `.format()` won't give you valid yaml in most cases (booleans are a case in point, and quite common in our config). A workaround would be to `json.dumps()` the values before the string substitution (which works because json is a subset of yaml). `yaml.dump()` doesn't work because it puts lots of new lines etc ... in the resulting string.

Reply:
Just double-checked: this does work for booleans on py3. For safety, I will switch to `json.dumps`.

Reply:
Tried switching to `json.dumps` and it wrapped everything in quotes, which made the regex not work. Also doesn't seem consistent with YAML. On my machine it works with the `.format()`, including with booleans.

Reply:
```python
In [25]: json.dumps(True)                                                                                                            
Out[25]: 'true'

In [26]: '{}'.format(True)                                                                                                           
Out[26]: 'True'

In [30]: yaml.dump(yaml.load('foo: true'))                                                                                           
Out[30]: '{foo: true}\n'

In [31]: yaml.dump(yaml.load('foo: True'))                                                                                           
Out[31]: '{foo: true}\n'
```

So booleans are supposed to be lower case, but pyyaml tolerates upper case when reading.

Also, I don't get any quotes
```
In [32]: 'foo: {}'.format(True)                                                                                                      
Out[32]: 'foo: True'

In [33]: 'foo: {}'.format(json.dumps(True))                                                                                          
Out[33]: 'foo: true'
```

Reply:
Ah, I thought you'd suggested `json.dumps` on the key as well as the value. The key is where I was getting the extra quotes. I've added `json.dumps` back in for the value. While it seems we can get away without `json.dumps` for booleans, it's likely safer for other data types.